### PR TITLE
Cleanup of main and optimizer initializations

### PR DESCRIPTION
**Pull request by nwittler**
**What**
General cleanup of the script based infrastructure.

**Why**
There was a lot of baggage from rigid old scripts. To make the software as a whole more extendable and usable, this needed to be cleaned up.

**How**
Moved relevant code from [c3.utils.parsers.py](https://github.com/q-optimize/c3/blob/dev/c3/utils/parsers.py) to the respective initialization methods or [c3.main.py](https://github.com/q-optimize/c3/blob/dev/c3/main.py)

**Open Questions**
- In calibration (C2) and model learning (C3) the configuration files and therefore the interface should be discussed
- In particular the handling of C2 communication with the experiment (simulated or real)

### CODE REVIEW

**Review by lazyoracle — DISMISSED** [review]
Ready to merge! 🚀

File: c3/main.py [hunk 1]
```diff
@@ -5,79 +5,82 @@
...
+            )
+        if "exp_opt_map" in cfg:
+            exp.pmap.set_opt_map(
+                [[tuple(par) for par in pset] for pset in cfg.pop("exp_opt_map")]
+            )
+
+        opt = optim_lib[optim_type](**cfg, pmap=exp.pmap)
         opt.set_exp(exp)
-        opt.set_created_by(opt_config)
+        opt.set_created_by(cfg)
```
Comment (nwittler) [inline c3/main.py:83]: This should be the location of the config file not the cfg dict.

**Review by fedroy — APPROVED** [review]
Just went thorugh it with @nwittler, up to hotfix of the storing the right config in main.py, the code is good for merging.

**Review by lazyoracle — APPROVED** [review]
:shipit:

**Pull request merged**